File: go/src/test/files/rules/detection/crossfile/CrossFileHookWrapper.go
package crossfile

import "crypto/md5"

// Wrapper defined in a SEPARATE file from its caller. Were Go to register a hook on this wrapper's
// parameter (as Java/Python do), that hook would have to resolve the call recorded while analyzing
// the caller file via the shared, whole-scan call stack — the leg the bucketing narrowing touches.
func HashIt(data []byte) [16]byte {
	return md5.Sum(data)
}

File: python/src/test/files/rules/resolve/imports/CrossFileHookWrapper.py
from cryptography.hazmat.primitives.asymmetric import ec


# Wrapper defined in a SEPARATE module from its caller. The hook that resolves `arg` is created
# while analyzing THIS file; it must match the `make_private_key(...)` call recorded while analyzing
# the caller module. This is the cross-file leg of the call-stack bucketing path.
def make_private_key(arg):
    return ec.generate_private_key(arg)
